plugins/pulp_docker/plugins/distributors/publish_steps.py

```diff
@@ -220,3 +222,126 @@ def process_main(self):
+        """
+        This method adds all the steps that are needed to accomplish an RPM rsync publish. This
+        includes:
+
+        Unit Query Step - selects units associated with the repo based on the date_filter and
+                          creates relative symlinks
+        Tag Generation Step  - creates relative symlinks for for all tags and writes out list file
+        Rsync Step (content units) - rsyncs content units from /var/lib/pulp/content to remote
+                                     server
+        Rsync Step (content) - rsyncs symlinks from working directory to remote server
+
+        :param date_filter:  Q object with start and/or end dates, or None if start and end dates
+                             are not provided
+        :type date_filter:  mongoengine.Q or types.NoneType
+        :param config: distributor configuration
+        :type config: pulp.plugins.config.PluginCallConfiguration
+        :return: None
+        """
+        remote_repo_path = configuration.get_repo_relative_path(self.repo, self.config)
+
+        unit_info = {constants.IMAGE_TYPE_ID: {'extra_path': [''], 'model': models.Image},
+                     constants.MANIFEST_TYPE_ID: {'extra_path': ['manifests'],
+                                                  'model': models.Manifest},
+                     constants.BLOB_TYPE_ID: {'extra_path': ['blobs'], 'model': models.Blob},
+                     constants.TAG_TYPE_ID: {'extra_path': ['manifests'], 'model': models.Tag}}
+
+        for type in DockerRsyncPublisher.REPO_CONTENT_TYPES:
+            unit_path = unit_info[type]['extra_path']
+            gen_step = RSyncFastForwardUnitPublishStep("Unit query step (things)",
+                                                       [unit_info[type]['model']],
+                                                       repo=self.repo,
+                                                       remote_repo_path=remote_repo_path,
+                                                       published_unit_path=unit_path)
+            self.add_child(gen_step)
+
+        tag_path = unit_info[constants.TAG_TYPE_ID]['extra_path']
+        tag_model = unit_info[constants.TAG_TYPE_ID]['model']
+        self.add_child(PublishTagsForRsyncStep("Generate tags step",
+                                               [tag_model],
+                                               repo=self.repo,
+                                               remote_repo_path=remote_repo_path,
+                                               published_unit_path=tag_path))
+
+        origin_dest_prefix = self.get_units_directory_dest_path()
+        origin_src_prefix = self.get_units_src_path()
+
+        self.add_child(RSyncPublishStep("Rsync step (content units)", self.content_unit_file_list,
+                                        origin_src_prefix, origin_dest_prefix,
+                                        config=config))
+
+        self.add_child(RSyncPublishStep("Rsync step (content)",
```

Comment:
A better title would help. As a user just looking at this title and the one above, it's not clear how they are different.